### DISCUSSION TITLE
Modifying shell activation script to resolve PATH issue in Nix installation

### DISCUSSION THREAD

**Discussion by pedorich-n** [jdx/mise#671] [General]
Hello!

#### Context
I am using Nix installation of `rtx`, from [nixpkgs](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/tools/misc/rtx/default.nix).

In Nix, packages are stored in the `/nix/store/...` directory, and they're symlinked to `~/.nix-profile/bin/...` directory, and then the `~/.nix-profile/bin/` folder is added to the `PATH` variable.

So for me, right now, the "real" `rtx` path is `/nix/store/4gc4pzmzdliim6fd978diis8pnxlshal-rtx-1.32.2/bin/rtx`, and it's exposed in Nix profile folder:
```shell
~ ❯ ls -go ~/.nix-profile/bin/ | grep rtx                    
lrwxrwxrwx 1  62 Jan  1  1970 rtx -> /nix/store/4gc4pzmzdliim6fd978diis8pnxlshal-rtx-1.32.2/bin/rtx
```

#### Problem
When `rtx`'s shell [activation script](https://github.com/jdxcode/rtx/blob/61d8ee7/src/shell/zsh.rs#L19-L21) runs, it can't find the `rtx` binary in `PATH` (because it looks for "real" `/nix/store/...` path) and exposes it again.

```Shell
~ ❯ echo $PATH 
...:/nix/store/4gc4pzmzdliim6fd978diis8pnxlshal-rtx-1.32.2/bin:/home/user/.nix-profile/bin:...
```

It's not a big problem, but it causes some inconvenience. 

</br>

Is there a way to change this behavior? I am happy to help with the PR, but I couldn't think of a good way to solve this without breaking the program for people who have `rtx` installed via `https://rtx.pub/install.sh` or just plain binary.

**Comment by jdx** [comment]
we could exclude that logic from doing anything if it's installed to /nix

**Reply by pedorich-n** [reply]
Personally, I would prefer a more "generic" solution (like a flag or more flexible setting), but if you think that's okay, then it's okay. 
I can work on it.

**Reply by jdx** [reply]
the problem with that is nix users would need to know to add it

**Reply by jdx** [reply]
we could probably also have a generic setting but I'm willing to wait until I hear of other use-cases

**Comment by pedorich-n** [comment]
Resolved in: https://github.com/jdxcode/rtx/pull/675

**Reply by dudicoco** [reply]
What is the advantage of using rtx within nix?
It's a package manager within a package manager.
